src/Controls/src/Core/Handlers/Items/iOS/CarouselViewLayout.cs

```diff
@@ -20,9 +20,9 @@ public CarouselViewLayout(ItemsLayout itemsLayout, CarouselView carouselView) :
 
 		public override void ConstrainTo(CGSize size)
 		{
-			//TODO: Should we scale the items 
-			var width = size.Width - _carouselView.PeekAreaInsets.Left - _carouselView.PeekAreaInsets.Right;
-			var height = size.Height - _carouselView.PeekAreaInsets.Top - _carouselView.PeekAreaInsets.Bottom;
+			// TODO: Should we scale the items 
+			var width = size.Width != 0 ? size.Width - _carouselView.PeekAreaInsets.Left - _carouselView.PeekAreaInsets.Right : 0;
```

Comment:
With the Container hidden, the size is zero, and in this case by subtracting PeekAreaInsets we had a negative item size that caused the exception.

The question is, if the CarouselView is hidden, should these calculations be avoided?

Reply:
I think is fine.